### PR TITLE
[RHEL6] rsyslog fixup

### PR DESCRIPTION
**Pull request by shawndwells**
- Align XCCDF to OVAL names
- Fix OVAL for rsyslog_files_groupownership

### CODE REVIEW

**Comment by redhatrises** [comment]
@shawndwells does this resolve #3?

**Comment by shawndwells** [comment]
On 4/8/15 4:21 PM, redhatrises wrote:
> @shawndwells <https://github.com/shawndwells> does this resolve #3 
> <https://github.com/OpenSCAP/scap-security-guide/issues/3>?

appears to! totally forgot about that old ticket.

File: RHEL/6/input/checks/rsyslog_files_groupownership.xml [hunk 1]
```diff
@@ -9,13 +8,25 @@
...
-  id="test_20155" version="1" />
+
+  <unix:file_test check="all" check_existence="all_exist" comment="Files that end in .log in /var/log owned by root" id="test_rsyslog_files_groupownership" version="1">
+    <unix:object object_ref="object_rsyslog_files_groupownership" />
+    <unix:state state_ref="state_rsyslog_files_groupownership" />
+  </unix:file_test>
+
+  <unix:file_object comment="Files that end in .log in /var/log" id="object_rsyslog_files_groupownership" version="2">
+    <unix:path>/var/log</unix:path>
+    <unix:filename operation="pattern match">.*log</unix:filename>
```
Comment (redhatrises) [inline RHEL/6/input/checks/rsyslog_files_groupownership.xml]: So what about logs such as ```/var/log/messages``` or ```/var/log/secure``` which do not end in ```.log```? What about logs/dirs that are not managed by rsyslog like ```Xorg.log``` or ```gdm.log``` or ```/var/log/btmp```?

**Comment by iankko** [comment]
@redhatrises, @shawndwells 
> So what about logs such as /var/log/messages or /var/log/secure which do not end in .log? What about logs/dirs that are not managed by rsyslog like Xorg.log or gdm.log or /var/log/btmp?

While I agree the above problem being an issue, it's not the only one with this OVAL check (and effort to scan services that might have recursively included configuration files). The main problem when trying to implement this rsyslog OVAL check being the ```IncludeConfig``` directive:

```
# grep Include /etc/rsyslog.conf 
# Include all config files in /etc/rsyslog.d/
$IncludeConfig /etc/rsyslog.d/*.conf
```

With the current version of ```oscap``` scanner in Red Hat Enterprise Linux 6 (supporting OVAL-5.10.1 language version), it is *not* possible to write the corresponding OVAL check in the way it would work correctly for all the possible configurations. This is actually the reason why the ```glob_to_regex()``` function has been introduced in OVAL-5.11 / OVAL-5.11.1:
&nbsp; &nbsp;  http://oval.mitre.org/language/version5.11/ovaldefinition/documentation/oval-definitions-schema.html#GlobToRegexFunctionType

In my opinion for this concrete case (and also cases of other services using recursively included configuration files) it is better not to have an OVAL check at all, or have OVAL check returning ```unknown``` result (IMHO current situation is better), than to have partially working OVAL check (which will report false positives even on non-compliant systems just due the restriction not being able to implement recursive config file inclusion correctly).

Therefore giving NACK for inclusion of this. If you want to have OVAL checks for these kind of rules implemented, ask the ```oscap``` tool provider to update ```openscap-utils``` | ```openscap-scanner``` package to support OVAL-5.11 / OVAL-5.11.1 language version in the downstream distribution.

**Pull request closed without merging**